src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -100,6 +115,108 @@ class StreamReader {
     [[nodiscard]] virtual auto decode_range(size_t begin_idx, size_t end_idx, bool use_filter) const
             -> DecodedResultsTsType = 0;
 
+    /**
+     * A templated function that implements `filter_log_events` for both
+     * `UnstructuredIrStreamReader` and `StructuredIrStreamReader`.
+     *
+     * @param log_level_filter
+     * @param log_events Derived class's log events.
+     * @param[out] filtered_log_event_map A reference to derived class's `FilteredLogEventsMap`
+     * that stores filtered result.
+     */
+    template <typename LogEvents>
+    static auto generic_filter_log_events(
+            FilteredLogEventsMap& filtered_log_event_map,
+            LogLevelFilterTsType const& log_level_filter,
+            LogEvents const& log_events
```

Comment:
```suggestion
    template <typename LogEvent>
    static auto generic_filter_log_events(
            FilteredLogEventsMap& filtered_log_event_map,
            LogLevelFilterTsType const& log_level_filter,
            std::vector<LogEventWithFilterData<LogEvent>> const& log_events
```

This is more explicit with the requirements for the type.